go/storage/mkvs/urkel/fuzz/tree.go

```diff
@@ -40,81 +69,109 @@ func (t *TreeFuzz) Insert(ctx context.Context, key []byte, value []byte) int {
 	}
 
 	t.reference[string(key)] = value
-	t.history = append(t.history, &mkvsTests.Op{Op: mkvsTests.OpInsert, Key: key, Value: value})
+
+	t.commitRemote(ctx)
+
 	return 0
 }
 
-func (t *TreeFuzz) Get(ctx context.Context, key []byte) int {
-	if len(key) == 0 {
-		// Ignore zero-length keys as they are invalid.
-		return -1
+func (t *TreeFuzz) get(ctx context.Context, tree mkvs.Tree, key []byte) {
+	if tree == nil {
+		return
 	}
 
-	value, err := t.inner.Get(ctx, key)
+	value, err := tree.Get(ctx, key)
 	if err != nil {
 		t.fail("Get failed: %s", err)
 	}
 
 	t.assertCorrectValue(key, value)
-
-	return 0
 }
 
-func (t *TreeFuzz) RemoveExisting(ctx context.Context, key []byte) int {
+func (t *TreeFuzz) Get(ctx context.Context, key []byte) int {
 	if len(key) == 0 {
 		// Ignore zero-length keys as they are invalid.
 		return -1
 	}
 
-	value, err := t.inner.RemoveExisting(ctx, key)
+	t.history = append(t.history, &mkvsTests.Op{Op: mkvsTests.OpGet, Key: key, Value: t.reference[string(key)]})
+
+	t.get(ctx, t.remote, key)
```

Comment:
how does this work usually? we keep resetting it to a tree with `.inner`'s roothash, but with nil database

Reply:
Yeah there is no database, the trees are only in-memory. So the remote tree uses the inner tree as a remote syncer, so `remote <-- ReadSyncer --> inner`. I'll add some comments to explain it.

Reply:
how does `remote` get a reference to `inner`?